keylime/secure_mount.py

```diff
@@ -88,10 +95,14 @@ def mount():
 def umount():
     """Umount all the devices mounted by Keylime."""
 
-    # Make sure we leave tmpfs dir empty even if we did not mount it or
-    # if we cannot unmount it. Ignore errors while deleting. The deletion
-    # of the 'secure' directory will result in an error since it's a mount point.
-    shutil.rmtree(mount(), ignore_errors=True)
+    secdir = get_secdir()
+    if not config.MOUNT_SECURE:
```

Comment:
Can you join both branches? `if check_mounted(secdir) or not config.MOUNT_SECURE:`

Reply:
Will do. I will just switch the order to not to check the mount only if MOUNT_SECURE is not used.